File: OOP Lab/Lab 03/Stack.java
import java.util.Scanner;

class Stack
{

    private int tos;
    private int stack[];
    public final int UNDERFLOW = -2147483648;    // INT_MIN

    Stack ()
    {
        tos = -1;
        stack = new int [0];
    }

    Stack (int size)
    {
        this.tos = -1;
        this.stack = new int [size];
    }

    boolean isFull ()
    {
        return (this.tos == this.stack.length - 1);
    }

    boolean isEmpty ()
    {
        return (this.tos == -1);
    }

    void push (int ele)
    {
        if (this.isFull())
        {
            System.err.println("Stack Overflow");
            return;
        }
        this.stack[++this.tos] = ele;
    }
    int pop ()
    {
        if (this.isEmpty())
        {
            System.err.println("Stack Underflow");
            return UNDERFLOW;
        }
        return this.stack[this.tos--];
    }
    void display ()
    {
        System.out.println("Stack now is");
        for (int i = 0; i <= this.tos; ++i)
            System.out.println(this.stack[i]);
        System.out.println();
    }
    

    public static void main (String [] args)
    {
        Scanner userEntry = new Scanner (System.in);
        System.out.println("Enter the number of elements");
        int size = userEntry.nextInt();
        
        Stack stack = new Stack (size);
        
        char choice;

        do
        {
            System.out.println("Enter 1 to push, 2 to pop, 3 to show, 0 to exit program");
            choice = userEntry.next().charAt(0);
            if (choice == '1')
            {
                System.out.println("Enter element to be added");
                int ele = userEntry.nextInt();
                stack.push(ele);
                stack.display();
            }
            else if (choice == '2')
            {
                int item = stack.pop();
                if (item != stack.UNDERFLOW)
                {
                    System.out.println("Deleted element is " + item);
                    stack.display();
                }
            }
            else if (choice == '3')
            {
                stack.display();
            }
            else
                System.exit(0);
        } while (choice == '1' || choice == '2' || choice == '3');
    }
}

File: DS Lab/DS Lab programs by week/Week 4/Stack.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 10
#define UNDERFLOW '\0'

void push (char *stack, int ele, int *top)
{
	if (*top == SIZE - 1)
	{
		printf("STACK OVERFLOW\n");
		return;
	}
	(*top) += 1;
	*(stack + (*top)) = ele;
}

char pop (char *stack, int *top)
{
	if (*top == -1)
	{
		printf("STACK UNDERFLOW\n");
		return UNDERFLOW;
	}
	return *(stack + ((*top)--));
}

void display (char *stack, int top)
{
	char *i;
	for (i = stack; i <= stack + top; ++i)
		printf("%c\n", *i);
}

int main()
{
	int top = -1;
	char *stack = (char *)malloc(sizeof(char *));
	int flag;
	
	printf("1 to push, 2 to pop, 3 to display, 0 to exit\n");
	scanf("%d", &flag);
	while (flag == 1 || flag == 2 || flag == 3)
	{
		if (flag == 1)
		{
			char ele;
			printf("Enter element to push \n");
			scanf("%s", &ele);
			push(stack, ele, &top);
			printf("Now stack is \n");
			display(stack, top);
		}
		else if (flag == 2)
		{
			char ele = pop(stack, &top);
			if (ele == UNDERFLOW)
			{
				exit(0);
			}
			printf("Deleted element = %c\n", ele);
			printf("Now stack is \n");
			display(stack, top);
		}
		else if (flag == 3)
		{
			printf("Now stack is \n");
			display(stack, top);
		}
		else
			exit(0);
		printf("1 to push, 2 to pop, 3 to display, 0 to exit\n");
		scanf("%d", &flag);
	}
	return 0;
}
